lredis/protocol.lua

```diff
@@ -36,43 +39,81 @@ local function encode_inline(arg)
 	return table.concat(arg, " ", 1, n+1)
 end
 
+-- Encode and send a redis command.
+local function send_command(file, arg)
+	local req, err_code = encode_request(arg)
+	if not req then
+		return nil, err_code
+	end
+	req, err_code = file:write(req)
+	if not req then
+		return nil, err_code
+	end
+	req, err_code = file:flush()
+	if not req then
+		return nil, err_code
+	end
+
+	return true
+end
+
 -- Parse a redis response
 local function read_response(file, new_status, new_error, string_null, array_null)
-	local line = assert(file:read("*l"))
-	assert(line:sub(-1, -1) == "\r", "invalid line ending")
+	local line, err_code = file:read("*l")
+	if not line then
+		return nil, err_code or "EOF reached"
```

Comment:
Sometimes this is a number, sometimes a string; need to be consistent on error types

Reply:
If ```file:read()``` fails then the default ```cqueues``` handler may ```assert()``` with a numeric code.  If ```file:read()``` returns something unexpected then your code ```assert()```'s a string.  There is no further inconsistency introduced by this change.